frontend/pages/views/MapPageView.js

```diff
@@ -0,0 +1,285 @@
+import React from "react";
+import {Container, Row, Col} from "react-bootstrap";
+import Map, {MAPSQUARE_HEIGHT, MAPSQUARE_WIDTH} from "../../components/ParisMap";
+import LoadingPage from "../LoadingPage";
+import Logo from "../../components/Logo";
+import Legend from "../../components/Legend";
+import {GeoJSON, Popup, Rectangle} from "react-leaflet";
+import * as PropTypes from "prop-types";
+
+function MapPageEntryLogo(){
+    return (<div>
+            <Container id="map-page-title">
+                <Row>
+                    <Col>
+                        <Logo id="site-logo" top={22} left={-1.5} logo_type={"title-logo"}/>
+                        <h2 className="page-title">Map</h2>
+                    </Col>
+                </Row>
+            </Container>
+            </div>
+    );
+}
+
+function MapSquareList(props){
+    //Given an arrondissiment, only keeps the map squares that actually contain photos
+    const mapSquareNumbers = props.arrondissementData[props.arrondissementNumber-1]["map_square_numbers"].filter(
+        number => props.filledMapSquares.has(number));
+
+    if(mapSquareNumbers.length === 0)
+        return <span>No Map Squares!</span>;
+
+    //Returns a comma delimited list of the map square numbers in a arrondissement. Each number
+    //links to corresponding map page
+    const displayedMapSquares = mapSquareNumbers.map((map_square_number, i) => {
+        const link = "/map_square/"+map_square_number+"/";
+        return (<span key={map_square_number}>
+                        <a href={link} className="info-list-link">{map_square_number}</a>
+                        {mapSquareNumbers[i+1] ? ", " : ""}
+                </span>);
+    });
+
+    return(<span>{displayedMapSquares}</span>);
+}
+
+function densityOverlay(mapData) {
+    const sortedMapData = Object.values(mapData)
+    .sort((a, b) => a.num_photos - b.num_photos);
+    // Gets the max number of photos in a single square out of all squares to form buckets later
+    const maxNumOfPhotos = sortedMapData && sortedMapData.length
+        ? sortedMapData[sortedMapData.length - 1].num_photos
+        : 0;
+    // Creating 5 buckets based on lowest to highest number of photos per square
+    const twentyPctMax = Math.round(0.2 * maxNumOfPhotos);
+    const fortyPctMax = Math.round(0.4 * maxNumOfPhotos);
+    const sixtyPctMax = Math.round(0.6 * maxNumOfPhotos);
+    const eightyPctMax = Math.round(0.8 * maxNumOfPhotos);
+    const buckets = [0, twentyPctMax, twentyPctMax + 1,
+        fortyPctMax, fortyPctMax + 1,
+        sixtyPctMax, sixtyPctMax + 1,
+        eightyPctMax, eightyPctMax + 1, maxNumOfPhotos];
+    const layer = (<>
```

Comment:
no need to initialize variable if you're just going to return it. Can return this whole clause instead.